BlazorClient/Pages/Play.razor.cs

```diff
@@ -0,0 +1,265 @@
+        }
+
+        if (_gameState == GameState.InProgress && _alive)
+        {
+            if (_currentRound == _lastRenderedRound)
+            {
+                return;
+            }
+
+            if (_width <= 0 || _height <= 0)
+        {
+            throw new InvalidOperationException($"Width: {_width}, Height: {_height}");
+        }
+        if (_boardSize.Width <= 0 || _boardSize.Height <= 0)
+        {
+            throw new InvalidOperationException($"boardWidth: {_boardSize.Width}, boardHeight: {_boardSize.Height}");
+        }
+
+        _lastRenderedRound = _currentRound;
+
+            await _context.ClearRectAsync(0, 0, _width, _height);
+
+            var xscale = _width / _boardSize.Width;
+            var yscale = _height / _boardSize.Height;
+            foreach (var p in _players)
+            {
+                if (p.Id == _id)
+                {
+                    await DrawPlayer(p, _blueSquare, _darkBlueSquare, xscale, yscale);
+                }
+                else if (p.HumanControlled)
+                {
+                    await DrawPlayer(p, _orangeSquare, _darkOrangeSquare, xscale, yscale);
+                }
+                else
+                {
+                    await DrawPlayer(p, _greenSquare, _darkGreenSquare, xscale, yscale);
+                }
+            }
+
+            foreach (var b in _berries)
+            {
+                await _context.DrawImageAsync(this._redSquare, b.X * xscale, b.Y * yscale, xscale, yscale);
+            }
+
+            await OutputTextAsync($"Score: {_score}", clear: false, 5, 50);
+        }
+    }
+
+    async Task DrawPlayer(PlayerState player, ElementReference headColor, ElementReference tailColor, int width, int height)
+    {
+        if (_context is null)
+        {
+            throw new InvalidOperationException($"'{nameof(_context)}' shouldn't be null.");
+        }
+
+        var head = player.Body[0];
+        await _context.DrawImageAsync(headColor, head.X * width, head.Y * height, width, height);
+        foreach (var b in player.Body.Skip(1))
+        {
+            await _context.DrawImageAsync(tailColor, b.X * width, b.Y * height, width, height);
+
+        }
+    }
+
+    void UpdatePlayerState(IList<PlayerState> players, IEnumerable<Point> berries)
+    {
+        this._players = players;
+        this._berries = berries;
+        this._currentRound++;
+    }
+
+    [JSInvokable]
+    public async ValueTask OnKeyDown(int keyCode)
+    {
+        if (_hubConnection is null)
+        {
+            return;
+        }
+
+        if (keyCode == (int)Keys.Left)
+        {
+            await _hubConnection.InvokeAsync("TurnLeft");
+        }
+        else if (keyCode == (int)Keys.Right)
+        {
+            await _hubConnection.InvokeAsync("TurnRight");
+        }
+    }
+
+    [JSInvokable]
+    public ValueTask OnResize(int width, int height)
+    {
+        _width = width;
+        _height = height;
+        return ValueTask.CompletedTask;
```

Comment:
This appears to be the thing that prevents GameLoop from being called.  Either making this async (with a warning), or void returning fixes it, but AFAICT there were no diagnostics that it caused a problem :(